### PR TITLE
Join Section | send selected data

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
### Issue

to be merged only after: https://github.com/Real-Dev-Squad/website-backend/pull/835

The join section converts complete local storage to JSON and sends it to the backend via fetch call.
This can cause issues as some users might have some extra data in their local storage that might be sent along with the required data which is not accepted by the backend thus this PR serves the purpose to solve the issue

### CODE REVIEW

File: js/join.js [hunk 1]
```diff
@@ -314,11 +343,11 @@ submit.addEventListener('click', async () => {
     headers: {
       'Content-Type': 'application/json',
     },
-    body: data,
+    body: JSON.stringify(sendJoinData()),
   })
     .then((res) => res.json())
     .then((res) => {
-      if (res.statusCode !== 200) {
+      if (res.message != 'User Data Added Succesfully') {
```
Comment (RitikJaiswal75) [inline js/join.js:350]: statusCode is not being sent from backend

Comment (RitikJaiswal75) [reply js/join.js:350]: Will make changes in nest pr

File: js/join.js [hunk 2]
```diff
@@ -243,6 +243,36 @@ function previewFiller() {
...
+    introduction,
+    skills,
+    college,
+    forFun,
+    whyRds,
+    foundFrom,
+    funFact,
+  };
+  return data;
+}
```
Comment (ankushdharkar) [inline js/join.js:274]: Could we generate it from a constants array please?
`['firstName', 'lastName', 'city', 'state'...]`

File: js/join.js [hunk 3]
```diff
@@ -314,11 +343,11 @@ submit.addEventListener('click', async () => {
     headers: {
       'Content-Type': 'application/json',
     },
-    body: data,
+    body: JSON.stringify(sendJoinData()),
   })
     .then((res) => res.json())
     .then((res) => {
-      if (res.statusCode !== 200) {
```
Comment (ankushdharkar) [inline js/join.js:321]: Why are we getting rid of the status code here? This would name our error messaging inflexible and unmaintainable .

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: js/join.js (hunk 3 above)
Comment (RitikJaiswal75) [reply js/join.js:321]: I used statusCode instead of `status` made the changes

File: js/join.js [hunk 4]
```diff
@@ -106,14 +106,16 @@ function fetchSavedDetails() {
...
   })
-    .then((res) => res.json())
     .then((res) => {
-      window.localStorage.setItem('firstName', res.first_name);
-      window.localStorage.setItem('lastName', res.last_name);
       if (res.statusCode === 401) {
         alert('You are not logged in! Redirecting you to login.');
         location.href = GITHUB_OAUTH;
       }
+      res.json();
```
Comment (ankushdharkar) [inline js/join.js:114]: !

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

**Review by ankushdharkar — APPROVED** [review]

**Pull request merged**